web/Database/PropertyDbModel.cs

```diff
@@ -39,6 +39,11 @@ public class PropertyDbModel
     public bool? AcceptsNotEET { get; set; }
     public bool? AcceptsWithoutGuarantor { get; set; }
 
-    // Rent, deposits, and duration
+    // Rent, deposits, availability and duration
     public int? Rent { get; set; }
+
+    public string Availability { get; set; } = "Draft";

```

Comment:
It would be nice to have constants for these values. you could add e.g.

```cs
public const string AVAILABILITY_DRAFT = "Draft";
```
to this class and then use that. That prevents typos and makes it easier to change them in the future if we want to.

Reply:
Yes.. there are four ways to do "enums" with our stack:

1. what we're doing, just store a string and hope nobody puts bad values in
2. by default if you make it an Enum EntityFramework will store the int representation (i.e. the first thing in the enum is 1, second is 2). That's fine, but it means reordering the enum in code without manually updating the column means it gets out of sync, and you can't tell what it means from the database alone. It is slightly more efficient, though that's not a big deal here.
3. you can use https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.storage.valueconversion.enumtostringconverter-1?view=efcore-6.0 to store it as a string instead. Could also optionally add a `CHECK` constraint to the table to make sure there are only those values in it
4. You can have a separate table and FK to it. So have an AvailabilityState table with a number and the enum meaning `(1, "Available")`, and then store the number in the Property table and add a foreign key to AvailabilityState to make sure it can't get out of sync. That's more work, but is otherwise "best" in that it enforces consistency while keeping the database meaningful without the code and allowing you to change enums by updating the AvailabilityState table

3 probably makes the most sense here really if you want to try it in another PR, though Microsoft don't seem to have bothered documenting how to use it, so you'll have to Google.

Reply:
I don't think I'll bother trying it in a separate PR, what we've already done seems fine to me - I was mostly just curious, since I haven't worked with enums before, and when I've tried to google them things haven't made that much sense. Thanks.